homeassistant/components/enphase_envoy/config_flow.py

```diff
@@ -89,24 +89,41 @@ async def async_step_zeroconf(
         self, discovery_info: zeroconf.ZeroconfServiceInfo
     ) -> ConfigFlowResult:
         """Handle a flow initialized by zeroconf discovery."""
+        current_hosts = self._async_current_hosts()
+        _LOGGER.debug(
+            "Zero config ip%s processing %s, current hosts: %s",
+            discovery_info.ip_address.version,
+            discovery_info.host,
+            current_hosts,
+        )
         if discovery_info.ip_address.version != 4:
             return self.async_abort(reason="not_ipv4_address")
         serial = discovery_info.properties["serialnum"]
         self.protovers = discovery_info.properties.get("protovers")
         await self.async_set_unique_id(serial)
         self.ip_address = discovery_info.host
         self._abort_if_unique_id_configured({CONF_HOST: self.ip_address})
+        _LOGGER.debug(
+            "Zero config ip %s , fw %s, no existing entry with serial %s",
+            self.ip_address,
+            self.protovers,
+            serial,
+        )
         for entry in self._async_current_entries(include_ignore=False):
             if (
                 entry.unique_id is None
                 and CONF_HOST in entry.data
                 and entry.data[CONF_HOST] == self.ip_address
             ):
+                _LOGGER.debug(
+                    "Zero config update envoy with this ip and blank serial in unique_id",
```

Comment:
Should we include any entry data in this log line? unique_id/host/serial (or lack thereof)?

Reply:
That was my original plan, until I realized its using an old entry with same ip and no unique_id in the entry. When the code gets here there's already 2 entries in the log like:

```txt
Zeroconf ip 4 processing 192.168.3.104, current hosts: set()
Zeroconf ip 192.168.3.104 , fw 8.2.127, no existing entry with serial 122302045041
```

which covered the entry data already. And if it is a new zeroconf detection followed by:
```txt
Zeroconf ip 192.168.3.104 to step user
```

For completeness, an ip switch would only log something like:
```txt
Zeroconf ip 4 processing 192.168.3.8, current hosts: {'192.168.3.104'}
Requesting https://192.168.3.8/info with timeout Timeout(connect=10.0, read=45.0, write=10.0, pool=10.0)
```